.github/workflows/binary-compatibility-checks.yml

```diff
@@ -23,28 +26,31 @@ jobs:
         scalaVersion: [ "2.12", "2.13" ]
     steps:
       - name: Checkout
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3.1.0
         with:
           # we don't know what commit the last tag was it's safer to get entire repo so previousStableVersion resolves
           fetch-depth: 0
 
+      - name: Cache Coursier cache
+        uses: coursier/cache-action@v6.4.0
+
       - name: Set up JDK 11
-        uses: olafurpg/setup-scala@v10
+        uses: coursier/setup-action@v1.3.0
         with:
-          java-version: adopt@1.11
-
-      - name: Cache Coursier cache
-        uses: coursier/cache-action@v6.2
+          jvm: adopt:1.11
 
       - name: Report MiMa Binary Issues
-        run: sbt -jvm-opts .jvmopts-ci "+~ ${{ matrix.scalaVersion }} mimaReportBinaryIssues"
+        run: |-
+          mv .jvmopts-ci .jvmopts
+          sbt "+~ ${{ matrix.scalaVersion }} mimaReportBinaryIssues"
 
       - name: Check correct MiMa filter directories
-        run: sbt -jvm-opts .jvmopts-ci checkMimaFilterDirectories
+        run: |
+          sbt checkMimaFilterDirectories
 
       - name: Email on failure
         if: ${{ failure() }}
-        uses: dawidd6/action-send-mail@v3
+        uses: dawidd6/action-send-mail@6063705cefe50cb915fc53bb06d4049cae2953b2
```

Comment:
Should the other ones also be using explicit hashes for security reasons? (Also maybe a comment with the version number for us humans?)

Reply:
I did it for the "privately owned" as I hope the others have enough eyes on them to not turn vicious. But you might be right to go all in.

Reply:
Ok, maybe that is enough.